edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -210,10 +210,21 @@ async Task<Twin> GetTwinFunc()
             catch (Exception e)
             {
                 Events.ErrorGettingTwin(e);
+
+                if (!forceNewModuleClient && this.RetryWithNewModuleClient(e))
+                {
+                    return await this.GetTwinFromIoTHub(true);
+                }
+
                 return Option.None<Twin>();
             }
         }
 
+        bool RetryWithNewModuleClient(Exception e)
```

Comment:
Nit: remove this method, not very useful

Reply:
Sure. I will do inline.